application_sdk/events/base.py

```diff
@@ -0,0 +1,143 @@
+"""Base event classes and models.
+
+This module contains the base classes and models for all events in the application.
+"""
+
+from abc import ABC
+from enum import Enum
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, Field
+
+from application_sdk.constants import APPLICATION_NAME
+
+
+class EventTypes(Enum):
+    """Enumeration of event types."""
+
+    APPLICATION_EVENT = "application_event"
+
+
+class ApplicationEventNames(Enum):
+    """Enumeration of application event names."""
+
+    WORKFLOW_END = "workflow_end"
+    WORKFLOW_START = "workflow_start"
+    ACTIVITY_START = "activity_start"
+    ACTIVITY_END = "activity_end"
+    WORKER_CREATED = "worker_created"
+
+
+class WorkflowStates(Enum):
+    """Enumeration of workflow states."""
+
+    UNKNOWN = "unknown"
+    RUNNING = "running"
+    COMPLETED = "completed"
+    FAILED = "failed"
+
+
+class EventMetadata(BaseModel):
+    """Metadata for events.
+
+    Attributes:
+        application_name: Name of the application the event belongs to.
+        event_published_client_timestamp: Timestamp when the event was published.
+        workflow_type: Type of the workflow.
+        workflow_id: ID of the workflow.
+        workflow_run_id: Run ID of the workflow.
+        workflow_state: State of the workflow.
+        activity_type: Type of the activity.
+        activity_id: ID of the activity.
+        attempt: Attempt number for the activity.
+        topic_name: Name of the topic for the event.
+    """
+
+    application_name: str = Field(init=True, default=APPLICATION_NAME)
+    event_published_client_timestamp: int = Field(init=True, default=0)
+
+    # Workflow information
+    workflow_type: str | None = Field(init=True, default=None)
+    workflow_id: str | None = Field(init=True, default=None)
+    workflow_run_id: str | None = Field(init=True, default=None)
```

Comment:
workflow_metadata as `map<string, string>` (workflow_id, run_id), activity_metadata(activity_id, attempt) and metadata